OpenSim/Common/Component.cpp

```diff
@@ -134,11 +152,19 @@ void Component::finalizeFromProperties()
 }
 
 // Base class implementation of virtual method.
-// Call extendFinalizeFromProperties on all components
+// Call finalizeFromProperties on all subcomponents
 void Component::componentsFinalizeFromProperties() const
```

Comment:
@tkuchida the sucbomponents list is about ownership. It is an invalid tree if the child owns the parent or any other component in its ancestry. I will add a check so that this isn't attempted. It is a good point that a component to be adopted cannot already by a component in the tree. There should be an exception for that. Thanks.